queries.sql

```diff
@@ -25,4 +25,93 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+-- Writing transaction for the following every transaction statement:
+
+-- Fist transaction
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. 
+-- Verify that change was made. 
+-- Then roll back the change and 
+-- verify that the species columns went back to the state before the transaction.
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Second transaction
+
+-- Inside a transaction:
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+-- Commit the transaction.
+-- Verify that change was made and persists after commit.
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+-- Third transaction
+
+-- Inside a transaction:
+-- Delete all animals born after Jan 1st, 2022.
+-- Create a savepoint for the transaction.
+-- Update all animals' weight to be their weight multiplied by -1.
+-- Rollback to the savepoint
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+-- Commit transaction
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT savepoint_name;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO savepoint_name;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
```

Comment:
- [ ] Kindly update the query after rollback in that transaction to meet the [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Update%20all%20animals%27%20weights%20that%20are%20negative%20to%20be%20their%20weight%20multiplied%20by%20%2D1.). They asked only to update weights that have a negative value, so you need to use `where` clause.